backend/socket.io/index.js

```diff
@@ -20,7 +20,11 @@ const initSocket = (server) => {
 
 const socketEmit = (topic, sid, data) => {
   const socket = io.sockets.sockets[sid];
-  socket.emit(topic, data);
+  return new Promise((resolve) => {
+    socket.emit(topic, data, (res) => {
+      resolve(res.ok);
+    });
+  });
```

Comment:
It's anti-pattern from a web-socket perspective. Acknowledgment should not be tightly coupled.